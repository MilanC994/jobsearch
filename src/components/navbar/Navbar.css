.bootstrap-navbar {
  padding: 0px 20px !important;
  background-color: #fff;
}
.navbar-nav {
  float: none;
  margin: 0 auto;
  display: block;
  text-align: center;
}
.navbar-links {
  position: absolute;
  top: 16px !important;
  right: 0px;
}
.fixed-div {
  min-width: 180px !important;
  height: 0px !important;
}
.navbar-collapse input[type='text'] {
  line-height: 28px;
  font-size: 15px;
  min-width: 15%;
  width: 47%;
  margin-right: 10px;
  display: flex;
}
.navbar-collapse .form-group {
  width: calc(100% - 110px);
}
.navbar-collapse form {
  width: 100%;
}
.navbar-collapse button {
  width: 100px;
  margin-left: auto;
  margin-right: 0;
}
.user-info-list {
  font-weight: bold !important;
  border: none !important;
}
.user-info-list:hover {
  cursor: pointer;
}
.navbar-links .list-group .list-group-item {
  padding: 0.45rem 1.25rem !important;
}
.user-info-list:nth-child(2):hover {
  background-color: #007bff;
}
@media (max-width: 433px) {
  .logo {
    width: 100px !important;
    height: 60px !important;
  }
  .navbar-links:nth-child(2) {
    color: red !important;
  }
  .navbar-links {
    top: 20px !important;
  }
  .navbar-links > form > div > a:first-child {
    display: none;
  }
}
@media (max-width: 991px) {
  .navbar-collapse input[type='text'] {
    padding: 25px 5px;
    line-height: 28px;
    font-size: 15px;
    width: 100%;
    margin-right: 2px;
  }
  .navbar-collapse button {
    font-size: 15px;
    padding: 5px 5px;
    width: 100%;
  }
  .navbar-collapse .form-group {
    width: 100% !important;
    margin-bottom: 0;
  }
  .navbar-links {
    right: 10px;
  }
}
