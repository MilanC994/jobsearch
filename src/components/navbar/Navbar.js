import React from 'react'
import './Navbar.css'
import { Link } from 'react-router-dom'
import useNavbar from './useNavbar'
import JoobleLogo from '../../imgs/jooble_logo.png'

function Navbar({ home }) {
  const {
    username,
    onLogout: logout,
    keywordsField,
    locationField,
    setParameters,
  } = useNavbar()
  return (
    <nav className="bootstrap-navbar navbar navbar-expand-lg navbar-light">
      <span className="navbar-brand">
        <Link id="app-logo-link" to="/">
          <img
            src={JoobleLogo}
            width="150"
            height="60"
            className="align-top logo"
            alt="logo "
          />
        </Link>
      </span>
      {home !== true && (
        <div
          className="m-auto navbar-collapse navbar-collapse collapse show"
          id="basic-navbar-nav"
        >
          <form className="form-inline">
            <div className="form-group">
              <input
                placeholder="Keywords"
                type="text"
                className="form-control"
                {...keywordsField}
              />
              <input
                placeholder="Location"
                type="text"
                className="form-control"
                {...locationField}
              />
            </div>
            <button
              onClick={setParameters}
              type="button"
              className="btn btn-primary btn-md"
            >
              Find Jobs
            </button>
          </form>
        </div>
      )}
      <div className="fixed-div" />
      <div className="align-top navbar-links justify-content-end navbar-nav">
        {username ? (
          <div
            id="username-logout-list"
            className="list-group list-group-horizontal"
          >
            <div className="user-info-list list-group-item list-group-item-primary">
              {username}
            </div>
            <div
              className="user-info-list list-group-item list-group-item-info"
              onClick={logout}
            >
              Logout
            </div>
          </div>
        ) : (
          <form id="login-register-form" className="form-inline">
            <div role="group" className="mr-2 btn-group">
              <Link id="register-link" to="/register">
                <button
                  id="register"
                  className="mx-2 btn btn-outline-primary btn-md"
                >
                  Register
                </button>
              </Link>
              <Link id="login-link" to="/login">
                <button
                  type="button"
                  className="mx-2 btn btn-outline-secondary btn-md"
                >
                  Login
                </button>
              </Link>
            </div>
          </form>
        )}
      </div>
    </nav>
  )
}

export default Navbar
