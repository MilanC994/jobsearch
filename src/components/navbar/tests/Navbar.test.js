import React from 'react'
import Navbar from '../Navbar'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { mountToJson } from 'enzyme-to-json'
import { StaticRouter as Router } from 'react-router-dom'

const mockHistoryPush = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}))

const createStore = configureMockStore([thunk])

const setUp = initialState => {
  const store = createStore(initialState)

  const wrapper = mount(
    <Provider store={store}>
      <Router>
        <Navbar />
      </Router>
    </Provider>
  ).childAt(0)
  return wrapper
}

describe('Header test', () => {
  const initialState = {
    user: {
      user: {},
      loggedIn: false,
    },
    jobs: {
      params: {
        keywords: '',
        location: '',
      },
    },
  }

  test('Renders Register and Login Buttons when user is not logged in', () => {
    const wrapper = setUp(initialState)
    expect(mountToJson(wrapper)).toMatchSnapshot()
    const registerLoginList = wrapper.find('#login-register-form')
    const userNameLogout = wrapper.find('#username-logout-list')
    expect(registerLoginList.exists()).toBeTruthy()
    expect(userNameLogout.exists()).toBeFalsy()
  })
  test('Renders username and logout list  when user is  logged in', () => {
    const state = {
      ...initialState,
      user: {
        ...initialState.user,
        user: {
          ...initialState.user.user,
          username: 'testUser',
        },
      },
    }
    const wrapper = setUp(state)
    expect(mountToJson(wrapper)).toMatchSnapshot()
    const registerLoginList = wrapper.find('#login-register-form')
    const userNameLogout = wrapper.find('#username-logout-list')
    expect(userNameLogout.exists()).toBeTruthy()
    expect(registerLoginList.exists()).toBeFalsy()
  })
  test('Click on logo redirects to home page', () => {
    const state = {
      ...initialState,
      user: {
        ...initialState.user,
        user: {
          ...initialState.user.user,
          username: 'testUser',
        },
      },
    }
    const wrapper = setUp(state)

    const appLogo = wrapper.find('#app-logo-link').at(0)
    expect(appLogo.exists()).toBeTruthy()
    expect(appLogo.prop('to')).toEqual('/')
  })
  test('Click on register redirects to register page', () => {
    const wrapper = setUp(initialState)

    const registerLink = wrapper.find('#register-link').at(0)
    expect(registerLink.exists()).toBeTruthy()
    expect(registerLink.prop('to')).toEqual('/register')
  })
  test('Click on login redirects to login page', () => {
    const wrapper = setUp(initialState)

    const loginLink = wrapper.find('#login-link').at(0)
    expect(loginLink.exists()).toBeTruthy()
    expect(loginLink.prop('to')).toEqual('/login')
  })
})
