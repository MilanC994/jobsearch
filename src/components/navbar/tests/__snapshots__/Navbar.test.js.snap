// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test Renders Register and Login Buttons when user is not logged in 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Navbar>
      <nav
        className="bootstrap-navbar navbar navbar-expand-lg navbar-light"
      >
        <span
          className="navbar-brand"
        >
          <Link
            id="app-logo-link"
            to="/"
          >
            <LinkAnchor
              href="/"
              id="app-logo-link"
              navigate={[Function]}
            >
              <a
                href="/"
                id="app-logo-link"
                onClick={[Function]}
              >
                <img
                  alt="logo "
                  className="align-top logo"
                  height="60"
                  src="jooble_logo.png"
                  width="150"
                />
              </a>
            </LinkAnchor>
          </Link>
        </span>
        <div
          className="m-auto navbar-collapse navbar-collapse collapse show"
          id="basic-navbar-nav"
        >
          <form
            className="form-inline"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                error={null}
                name="keywords"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Keywords"
                type="text"
                value=""
              />
              <input
                className="form-control"
                error={null}
                name="location"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Location"
                type="text"
                value=""
              />
            </div>
            <button
              className="btn btn-primary btn-md"
              onClick={[Function]}
              type="button"
            >
              Find Jobs
            </button>
          </form>
        </div>
        <div
          className="fixed-div"
        />
        <div
          className="align-top navbar-links justify-content-end navbar-nav"
        >
          <form
            className="form-inline"
            id="login-register-form"
          >
            <div
              className="mr-2 btn-group"
              role="group"
            >
              <Link
                id="register-link"
                to="/register"
              >
                <LinkAnchor
                  href="/register"
                  id="register-link"
                  navigate={[Function]}
                >
                  <a
                    href="/register"
                    id="register-link"
                    onClick={[Function]}
                  >
                    <button
                      className="mx-2 btn btn-outline-primary btn-md"
                      id="register"
                    >
                      Register
                    </button>
                  </a>
                </LinkAnchor>
              </Link>
              <Link
                id="login-link"
                to="/login"
              >
                <LinkAnchor
                  href="/login"
                  id="login-link"
                  navigate={[Function]}
                >
                  <a
                    href="/login"
                    id="login-link"
                    onClick={[Function]}
                  >
                    <button
                      className="mx-2 btn btn-outline-secondary btn-md"
                      type="button"
                    >
                      Login
                    </button>
                  </a>
                </LinkAnchor>
              </Link>
            </div>
          </form>
        </div>
      </nav>
    </Navbar>
  </Router>
</StaticRouter>
`;

exports[`Header test Renders username and logout list  when user is  logged in 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <Navbar>
      <nav
        className="bootstrap-navbar navbar navbar-expand-lg navbar-light"
      >
        <span
          className="navbar-brand"
        >
          <Link
            id="app-logo-link"
            to="/"
          >
            <LinkAnchor
              href="/"
              id="app-logo-link"
              navigate={[Function]}
            >
              <a
                href="/"
                id="app-logo-link"
                onClick={[Function]}
              >
                <img
                  alt="logo "
                  className="align-top logo"
                  height="60"
                  src="jooble_logo.png"
                  width="150"
                />
              </a>
            </LinkAnchor>
          </Link>
        </span>
        <div
          className="m-auto navbar-collapse navbar-collapse collapse show"
          id="basic-navbar-nav"
        >
          <form
            className="form-inline"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                error={null}
                name="keywords"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Keywords"
                type="text"
                value=""
              />
              <input
                className="form-control"
                error={null}
                name="location"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Location"
                type="text"
                value=""
              />
            </div>
            <button
              className="btn btn-primary btn-md"
              onClick={[Function]}
              type="button"
            >
              Find Jobs
            </button>
          </form>
        </div>
        <div
          className="fixed-div"
        />
        <div
          className="align-top navbar-links justify-content-end navbar-nav"
        >
          <div
            className="list-group list-group-horizontal"
            id="username-logout-list"
          >
            <div
              className="user-info-list list-group-item list-group-item-primary"
            >
              testUser
            </div>
            <div
              className="user-info-list list-group-item list-group-item-info"
              onClick={[Function]}
            >
              Logout
            </div>
          </div>
        </div>
      </nav>
    </Navbar>
  </Router>
</StaticRouter>
`;
